### ISSUE TITLE
How to reproduce YOLOv8 experimental results (train from scratch) on coco(train2017/val2017)?

### ISSUE THREAD

**Issue by chihyuwang** [ultralytics/ultralytics#459]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Is there any command line or document for reproducing YOLOv8 experimental results (train from scratch) on coco(train2017/val2017)?

We can't reproduce YOLOv8 experimental results (train from scratch) on coco(train2017/val2017) when we use the hyperparameters in yolo/configs/default.yaml.

**Comment by chihyuwang** [comment]
Below is the reproducing yolov8-n YOLOv8 experimental code and results, we can't reproduce YOLOv8 official results. 

from ultralytics import YOLO
model = YOLO('yolov8n.yaml')
results = model.train(data='coco.yaml')  # train the model

The tensorboard results.

![image](https://user-images.githubusercontent.com/25482742/213678724-71087587-111c-4f99-b334-0c48dca96a57.png)
![image](https://user-images.githubusercontent.com/25482742/213678857-a7d4df85-4c9b-44b7-8e2c-51b2e43f4311.png)
![image](https://user-images.githubusercontent.com/25482742/213679853-b8298aa6-a8a9-4bb4-8304-5b4b191815fd.png)

How to reproduce the YOLOv8 official results(train from scratch)?

**Comment by AyushExel** [comment]
You need to train for more epochs. Train for 500 epochs and you should get simialr mAPs. The official reproducibility command will be released with the paper soon.

**Comment by minyong-cho** [comment]
@AyushExel
Hi, I saw this issue last week and I trained 500 epochs as you commented.
Actually, I got 36.6mAP, slightly different from the official result (37.3mAP).
And when I validated with the pre-trained model (`model = YOLO("yolov8n.pt")`), it also show a slightly different result (37.1) compared to the official result (37.3mAP).
Coult you tell me what am I missing?

Thank you very much.

**Comment by 523997931** [comment]
> @AyushExel Hi, I saw this issue last week and I trained 500 epochs as you commented. Actually, I got 36.6 mAP, slightly different from the official result (37.3 mAP). And when I validated with the pre-trained model (`model = YOLO("yolov8n.pt")`), it also show a slightly different result (37.1 mAP) compared to the official result (37.3 mAP). Could you tell me what am I missing?
> 
> Thank you very much.

hello,

**Comment by 523997931** [comment]
> @AyushExel Hi, I saw this issue last week and I trained 500 epochs as you commented. Actually, I got 36.6 mAP, slightly different from the official result (37.3 mAP). And when I validated with the pre-trained model (`model = YOLO("yolov8n.pt")`), it also show a slightly different result (37.1 mAP) compared to the official result (37.3 mAP). Could you tell me what am I missing?
> 
> Thank you very much.

hello,
recently, I also try to train coco, use yolov8n, but in my training process, the train loss can't decrease, and the val map is always 0.000..., I don't know why, do you meet this situation？ Or did you change some parameters to train? Thx

**Comment by minyong-cho** [comment]
No. I did not meet that situation with default parameters.

**Comment by glenn-jocher** [comment]
@lavenderlove52 to open Tensorboard, you must first save your training results in the format of an event file using a command that outputs the path to the saved event file. In the case of YOLOv8, this may be done using a command such as `tensorboard --logdir=<path-to-results-folder>`. Once this command is run, you can open Tensorboard by entering `<ip-address-of-server-running-tensorboard>:<port>` in your web browser. From here, you can view various metrics such as loss, mAP, and learning rate decay over different epochs and iterations. Tensorboard is a powerful tool for gaining insight into the performance of your model and diagnosing issues with your training process, so it is highly recommended to regularly monitor the training results in Tensorboard throughout the training process.

**Comment by FFF-fenghen** [comment]
I get the similar result too, yolov8n train on coco2017 from scratch, I got map 36.8, compare to 37.2, some hyps was set as followlling:
batchsize = 128,
epoch = 500,
device = 0,1,2,3
train on RTX3090
hypsno mention are set on defalut. Do I miss anything?

**Comment by glenn-jocher** [comment]
@FFF-fenghen hello! Thanks for trying out the YOLOv8 repo on Coco 2017. To achieve better results, you may want to try experimenting with hyperparameters such as learning rate, learning rate schedule, data augmentation, weight decay, and more. You may also want to consider training for a longer period of time than 500 epochs, depending on the size of your dataset and the complexity of your model. Additionally, you may consider using different models such as YOLOv5 or customizing your own model architecture to suit your specific use case and data. Good luck with your experimentation and please let us know if you have any further questions or concerns!

**Comment by CretuCalin** [comment]
> You need to train for more epochs. Train for 500 epochs and you should get simialr mAPs. The official reproducibility command will be released with the paper soon.

Is there any update regarding the reproducibility command?

**Comment by glenn-jocher** [comment]
@CretuCalin certainly, we appreciate your patience. We're currently working towards finalizing the reproducibility command for YOLOv8. However, we believe it's crucial to ensure its proper functioning before releasing it. Thus, we ask for a bit more patience on your end. We assure you; this update will certainly enhance the efficiency and accuracy of your YOLOv8 experiments. Stay tuned for further updates on this, as they will be posted in the repository's README or in the "Releases" section. Your understanding and support are greatly appreciated.

**Comment by ZouJiu1** [comment]
the training log helps a lot, thank you

**Comment by kazu-gra** [comment]
> You need to train for more epochs. Train for 500 epochs and you should get simialr mAPs. The official reproducibility command will be released with the paper soon.

Hello.
How long did it take you to learn 500 epochs of the yolov8 scratch model? I would also like to know what GPU you used to do that.
By the way, I will be using 3 x NVIDIA A100 (80G), can you give me an approximate time?

**Comment by glenn-jocher** [comment]
@kazu-gra hello there,

The total training time can vary significantly depending on many factors such as the specific dataset size, your chosen batch size, network architecture, etc. That said, learning 500 epochs on a YOLOv8 model is a substantial task.

Given your setup of 3 x NVIDIA A100 (80G), it's safe to say that this is a very high-end setup and would be expected to perform very well. However, providing an exact timeline is challenging without knowing the specifics of your project.

From experience, with a similar setup, one could expect a single epoch to take on the order of minutes. Therefore, for 500 epochs you can expect it to take several days. Of course, this should be considered a ballpark estimate.

Lastly, bear in mind that the computational efficiency is also highly correlated to the implementation details and how well the code is optimized to leverage the hardware capabilities. The Ultralytics team has put significant effort into ensuring that Ultralytics-based repositories like YOLOv8 are highly optimized in this regard.

Best of luck with your training! Let me know if you have any other questions.

**Comment by Sohaya1** [comment]
@minyong-cho Bonjour,
 Combien de temps a pris l'entraînement lorsque tu as utilisé 500 epochs ? As-tu travaillé avec Google Colab ? Merci d'avance.
